test/end-to-end/company.js

```diff
@@ -1933,4 +1933,294 @@ describe("Company endpoint", () => {
+            });
+        });
+
+        test("Offer should be updated", async () => {
+            await test_agent
+                .post("/auth/login")
+                .send(test_user_company)
+                .expect(HTTPStatus.OK);
+
+            const res = await test_agent
+                .put(`/company/${test_company._id}/edit`)
+                .send({
+                    name: "Changing Company",
+                    contacts: ["122"],
+                })
+                .expect(HTTPStatus.OK);
+
+            test_offer = await Offer.findById(test_offer._id);
+            expect(res.body).toHaveProperty("contacts", ["122"]);
+            expect(test_offer.ownerName).toEqual("Changing Company");
+            expect(test_offer.contacts).toEqual(["122"]);
+        });
+
+        describe("Should fail if company is blocked or disabled", () => {
+            test("Should fail if company is blocked (admin)", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                await Company.findByIdAndUpdate(test_company._id, { isBlocked: true });
+
+                const res = await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send({
+                        name: "Changing Blocked Company",
+                        bio: "As admin",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
+                    .expect(HTTPStatus.FORBIDDEN);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.COMPANY_BLOCKED);
+            });
+
+            test("Should fail if company is disabled (admin)", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                await Company.findByIdAndUpdate(test_company._id, { isDisabled: true });
+
+                const res = await test_agent
+                    .put(`/company/${test_company._id}/edit`)
+                    .send({
+                        name: "Changing Disabled Company",
+                        bio: "As admin",
+                        contacts: ["1"],
+                        logo: "http://awebsite.com/otherlogo.jpg",
+                    })
+                    .expect(HTTPStatus.FORBIDDEN);
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.COMPANY_DISABLED);
+            });
+
+            test("Should fail if company is blocked (user)", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_company)
+                    .expect(HTTPStatus.OK);
+
+                await Company.findByIdAndUpdate(test_company._id, { isBlocked: true });
```

Comment:
Same thing as above, since this is supposed to fail you can reuse the same company used in the other test: it is guaranteed to not change so there is no problem.